ui/src/components/directory/topbar/eclSearch/EclTree.vue

```diff
@@ -0,0 +1,397 @@
+import { ConceptAggregate, ConceptSummary, EntityReferenceNode, TreeParent, TTIriRef } from "@im-library/interfaces";
+import { IM, RDF, RDFS } from "@im-library/vocabulary";
+import { EntityService } from "@/services";
+import setupTree from "@/composables/setupTree";
+import { TreeNode } from "primevue/tree";
+import { toTitleCase } from "@im-library/helpers/StringManipulators";
+import { useToast } from "primevue/usetoast";
+
+const {
+  root,
+  expandedKeys,
+  selectedKeys,
+  createLoadMoreNode,
+  createTreeNode,
+  onNodeCollapse,
+  onNodeDblClick,
+  onNodeExpand,
+  onRowClick,
+  loadMore,
+  selectAndExpand,
+  scrollToHighlighted,
+  loadMoreChildren,
+  locateChildInLoadMore,
+  selectKey,
+  selectedNode
+} = setupTree();
+
+const toast = useToast();
+
+const dialogRef = inject("dialogRef");
+
+const rootConceptIri = ref("");
+const conceptAggregates: Ref<ConceptAggregate[]> = ref([]);
+const hoveredResult: Ref<ConceptSummary> = ref({} as ConceptSummary);
+const overlayLocation: Ref = ref({});
+const loading = ref(false);
+const totalCount = ref(0);
+const superiorsCount = ref(0);
+const pageSize = ref(20);
+
+const altTreeOP = ref();
+
+watch(
+  () => dialogRef.value.data.focus.iri,
+  async newValue => {
+    selectedKeys.value = {};
+    expandedKeys.value = {};
+    if (newValue) {
+      await getConceptAggregates(newValue);
+      for (const aggregate of conceptAggregates.value) {
+        await createTree(aggregate.concept, aggregate.children);
+      }
+    }
+  }
+);
+
+watch(loading, newValue => {
+  if (newValue) hidePopup(overlayLocation.value);
+});
+
+onMounted(async () => {
+  setRootConceptIri();
+  if (rootConceptIri.value) {
+    await getConceptAggregates(rootConceptIri.value);
+    for (const aggregate of conceptAggregates.value) {
+      await createTree(aggregate.concept, aggregate.children);
+    }
+    if (root.value.length < superiorsCount.value) {
+      root.value.push(
+        createLoadMoreNode(
+          createTreeNode(dialogRef.value.data.focus.iri, dialogRef.value.data.focus.name, [{ "@id": IM.CONCEPT, name: "Concept" }], false, null),
+          1,
+          superiorsCount.value
+        )
+      );
+    }
+    if (dialogRef.value.data.currentValue) {
+      await findSelected(dialogRef.value.data.currentValue.iri);
+    }
+  }
+});
+
+onBeforeUnmount(() => {
+  if (isObject(overlayLocation.value) && isArrayHasLength(Object.keys(overlayLocation.value))) {
+    hidePopup(overlayLocation.value);
+  }
+  dialogRef.value.close();
+});
+
+function setRootConceptIri() {
+  if (dialogRef.value && isObjectHasKeys(dialogRef.value, ["data"]) && isObjectHasKeys(dialogRef.value.data, ["focus"]))
+    rootConceptIri.value = dialogRef.value.data.focus.iri;
+}
+
+async function getConceptAggregates(iri: string): Promise<void> {
+  loading.value = true;
+  let superiors = [];
+  if (dialogRef.value.data.type === "property") {
```

Comment:
worth refactoring into an inline check and direct assign to avoid code duplication
const superiors = dialogRef.value.data.type === "property" 
                            ? await EntityService.getSuperiorPropertiesPaged(rootConceptIri.value, 1, pageSize.value);
                            : await EntityService.getSuperiorPropertyValuesPaged(rootConceptIri.value, 1, pageSize.value)

Reply:
Going to leave as is, not much duplication and getSuperiors require destructing into separate variables.